File: ui-preview/src/main/java/l/files/thumbnail/ImageThumbnailer.java
package l.files.thumbnail;

import android.content.Context;

import l.files.fs.Path;
import l.files.ui.base.graphics.Rect;
import l.files.ui.base.graphics.ScaledBitmap;

import static l.files.ui.base.graphics.Bitmaps.decodeScaledDownBitmap;

final class ImageThumbnailer implements Thumbnailer<Path> {

    @Override
    public boolean accepts(Path path, String mediaType) {
        return mediaType.startsWith("image/");
    }

    @Override
    public ScaledBitmap create(Path path, Rect max, Context context)
            throws Exception {
        return decodeScaledDownBitmap(path::newInputStream, max);
    }

}

File: ui-preview/src/main/java/l/files/thumbnail/ImageThumbnailer.kt
package l.files.thumbnail

import android.content.Context
import l.files.ui.base.graphics.Bitmaps
import l.files.ui.base.graphics.Rect
import java.nio.file.Files.newInputStream
import java.nio.file.Path

internal object ImageThumbnailer : Thumbnailer<Path> {

  override fun accepts(path: Path, type: String) =
    type.startsWith("image/")

  override fun create(input: Path, max: Rect, context: Context) =
    Bitmaps.decodeScaledDownBitmap({ newInputStream(input) }, max)
}
